src/components/ChannelPreviewMessenger.js

```diff
@@ -81,15 +81,36 @@ export const ChannelPreviewMessenger = themed(
       this.props.setActiveChannel(this.props.channel);
     };
 
+    renderAvatar = (otherMembers) => {
+      const { channel } = this.props;
+      if (channel.data.image)
+        return <Avatar image={channel.data.image} size={40} />;
+
+      if (otherMembers.length === 1)
+        return <Avatar image={otherMembers[0].user.image} size={40} />;
+
+      return <Avatar size={40} />;
+    };
+
     render() {
       const { channel } = this.props;
+      let name = channel.data.name;
+      let otherMembers = [];
+      if (!name) {
+        const members = Object.values(channel.state.members);
+        otherMembers = members.filter(
+          (member) => member.user.id !== this.props.client.userID,
+        );
+        name = otherMembers.map((member) => member.user.name).join(', ');
```

Comment:
maybe a fallback if there's no `user.name` just call it `untitled` or something

Reply:
Done!!